docs/rst/tutorials/tools/monitor_with_backend/code/monitor/src/monitor.cpp

```diff
@@ -0,0 +1,421 @@
+        , n_bins_(n_bins)
+        , t_interval_(t_interval)
+    {
+    }
+
+    ~Monitor()
+    {
+        StatisticsBackend::stop_monitor(monitor_id_);
+    }
+
+    bool init()
+    {
+        monitor_id_ = StatisticsBackend::init_monitor(domain_);
+        if (!monitor_id_.is_valid())
+        {
+            std::cout << "Error creating monitor" << std::endl;
+            return 1;
+        }
+
+        StatisticsBackend::set_physical_listener(&physical_listener_);
+
+        return true;
+    }
+
+    void run()
+    {
+        stop_.store(false);
+        std::cout << "Monitor running. Please press CTRL+C to stop the Monitor at any time." << std::endl;
+        signal(SIGINT, [](int signum)
+                {
+                    std::cout << "\nSIGINT received, stopping Monitor execution." << std::endl;
+                    static_cast<void>(signum);
+                    Monitor::stop();
+                });
+
+        while (!stop_.load())
+        {
+            stop_.load();
+
+            get_fastdds_latency_mean();
+            get_publication_throughput_mean();
+        }
+    }
+
+    static void stop()
+    {
+        stop_.store(true);
+    }
+
+
+    std::vector<StatisticsData> get_fastdds_latency_mean()
+    {
+        std::vector<StatisticsData> latency_data{};
+
+        std::vector<EntityId> topics = StatisticsBackend::get_entities(EntityKind::TOPIC);
+        EntityId chatter_topic_id = -1;
+        Info topic_info;
+        for (auto topic_id : topics)
+        {
+            topic_info = StatisticsBackend::get_info(topic_id);
+            if (topic_info["name"] == "rt/chatter" && topic_info["data_type"] == "std_msgs::msg::dds_::String_")
+            {
+                chatter_topic_id = topic_id;
+                break;
+            }
+        }
+
+        if (chatter_topic_id < 0)
+        {
+            return latency_data;
+        }
+
+        std::vector<EntityId> topic_datawriters = StatisticsBackend::get_entities(
+            EntityKind::DATAWRITER,
+            chatter_topic_id);
+        std::vector<EntityId> topic_datareaders = StatisticsBackend::get_entities(
+            EntityKind::DATAREADER,
+            chatter_topic_id);
+
+        std::chrono::system_clock::time_point now = std::chrono::system_clock::now();
+
+        latency_data = StatisticsBackend::get_data(
+            DataKind::FASTDDS_LATENCY,                                   // DataKind
+            topic_datawriters,                                           // Source entities
+            topic_datareaders,                                           // Target entities
+            n_bins_,                                                           // Number of bins
+            now - std::chrono::seconds(t_interval_),                               // t_from
+            now,                                                         // t_to
+            StatisticKind::MEAN);                                        // Statistic
```

Comment:
Fix comment indentation.

```suggestion
            DataKind::FASTDDS_LATENCY,                                   // DataKind
            topic_datawriters,                                           // Source entities
            topic_datareaders,                                           // Target entities
            n_bins_,                                                           // Number of bins
            now - std::chrono::seconds(t_interval_),                               // t_from
            now,                                                         // t_to
            StatisticKind::MEAN);                                        // Statistic
```